### ISSUE TITLE
Can't deserialize nested structs

### ISSUE THREAD

**Issue by yassinebenarbia** [dbus2/zbus#918]
TL;DR: I was attempting to send a nested structure over the bus but couldn't.

Basically I had an implementation that uses the  `interface` macro, and one of the methods returns a different nested structure (a struct that one of it's fields is also a struct), I've used the derive macro to implement the `Type`, `DeserializeDict` and `SerializeDict` and got an error when I tried to send it (return it from an implemented method) using this command:
`busctl --user call org.zbus.myapp /org/zbus/myapp org.zbus.myapp1 Show`
The error:
```
Call failed: invalid type: character `(`, expected `s`, `g`, `o` or `v`
```

When I've inspected the signature of the nested structure, I got the following using the `signature` method:
```(s(ay))```

This is a representation of the actual structure that I was trying to send
```Rust
struct Insider {
  nest: Vec<u8>,
}

struct Nested<'a> {
  first_thing: &'a str,
  insider: Insider,
}
```
Question:
- How can I send the nested structure?

**Comment by zeenix** [comment]
Thanks. Nested structs are not an issue, it's nested dictionaries through *Dict macros that has this limitation. It's a known issue though (https://github.com/dbus2/zbus/issues/312) so I'll close this as duplicate.
